### PR TITLE
add support for custom ordering

### PR DESCRIPTION
**Pull request by LukeAskew**
Allows users to add number prefixes to materials to control their ordering.

These files:

```
01-foo.html
01.01-bar.html
02-qux.html
```

Will list as:

```
Foo
Bar
Qux
```

And are still accessed as:

```
{{> foo}}
{{> bar}}
{{> qux}}
```

closes https://github.com/fbrctr/fabricator/issues/126

### CODE REVIEW

**Comment by LukeAskew** [comment]
shout out to @sethlopezme for his help on the regex https://github.com/fbrctr/fabricator-assemble/commit/2b69a6dc4683ee3a53d87556b95827a8db49c75b#diff-168726dbe96b3ce427e7fedce31bb0bcR539

**Comment by lyzadanger** [comment]
I'm super happy to see this! And I love that it is in `fabricator-assemble` so that it can be a simple version bump for my team. Rock. On.

File: README.md [hunk 1]
```diff
@@ -306,6 +306,26 @@ The content within these files can be accessed as such:
...
+You can manually order materials by prefixing the file name with numbers:
+
+```
+01-foo.html
+01.01-bar.html
+02-qux.html
+```
+
+This defines the order in which materials will appear in the side menu or other places the `materials.items` context is used.
+
```
Comment (lyzadanger) [inline README.md]: Would it be useful to update the `README` somewhere around here to mention that the `name` property for materials will also reflect the prefix-stripping? (I mean, I assume the `name` property is also cleaned up).

**Comment by tylersticka** [comment]
This looks awesome! :+1:

File: README.md (hunk 1 above)
Comment (LukeAskew) [inline README.md]: Good call out. I can add that.

**Pull request merged**